tools/cli/commands/rsync.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * External dependencies
+ */
+import fs from 'fs';
+import process from 'process';
+import chalk from 'chalk';
+import execa from 'execa';
+import inquirer from 'inquirer';
+import watch from 'node-watch';
+import tmp from 'tmp';
+/**
+ * Internal dependencies
+ */
+import { projectDir } from '../helpers/install.js';
+import { listProjectFiles } from '../helpers/list-project-files.js';
+import { allProjectsByType, dirs } from '../helpers/projectHelpers.js';
+import { runCommand } from '../helpers/runCommand.js';
+
+/**
+ * Entry point for the CLI.
+ *
+ * @param {object} argv - The argv for the command line.
+ */
+export async function rsyncInit( argv ) {
+	argv = await promptForPlugin( argv );
+	argv = await promptForDest( argv );
+	const sourcePluginPath = projectDir( `plugins/${ argv.plugin }` );
+
+	if ( true === argv.watch ) {
+		console.log( chalk.yellow( `Watching for changes in ${ argv.plugin } to rsync` ) );
+		watch( sourcePluginPath, { recursive: true }, ( eventType, fileName ) => {
+			if ( fileName && 'update' === eventType ) {
+				if ( argv.v ) {
+					console.log( chalk.yellow( `Changed detected in ${ fileName }...` ) );
+				}
+				rsyncToDest( sourcePluginPath, argv.dest );
```

Comment:
You might want to add some logic in here so if a `pnpm install` or something changes 1000 files it doesn't wind up running rsync 1000 times in parallel.

Reply:
As mentioned above, going to remove watch from this PR